starport/services/network/networkchain/networkchain.go

```diff
@@ -79,15 +80,37 @@ func SourceRemoteHash(url, hash string) SourceOption {
 	}
 }
 
+// Launch represents the launch of a chain on SPN
 type Launch struct {
 	ID          uint64
 	ChainID     string
 	SourceURL   string
 	SourceHash  string
 	GenesisURL  string
 	GenesisHash string
+	CampaignID  uint64
 }
 
+// ParseLaunch parses a chain launch from SPN and returns a launch object
+func ParseLaunch(chainLaunch launchtypes.Chain) Launch {
```

Comment:
networkchain seems like does not need to know more about what is a launch, it might make sense to move this func to the network pkg instead and use the LaunchInfo type there: https://github.com/tendermint/starport/blob/3b0da49/starport/services/network/network.go#L51

Also maybe it's better to not add `CampaignID` to `Launch` inside network builder because networkbuilder does not need to know about it.

Instead `LaunchInfo` in the network pkg can embed networkchain.Launch and add the extra `CampaignID` field.

Reply:
> Also maybe it's better to not add CampaignID to Launch inside network builder because networkbuilder does not need to know about it.

Yes I add some confusion with the `Launch` structure. This is all the necessary information to build the genesis of the chain. I would rename the structure LaunchInformation

Reply:
I actually had some confusion with the `Launch` object in `networkchain` I though it would be used for the chain genesis construction logic after but as far as I can see everything would be done through the `Chain` structure. If we only use `Launch` for the initialization I would preferably consider having the `Launch` structure defined in `network` with everything that would be needed like campaign and pass information to `networkchain` for initialization through parameters
```
// SourceLaunch returns a source option for initializing a chain from a launch
func SourceLaunch(
	chainID,
	sourceURL,
	sourceHash,
	genesisURL,
	genesisHash string,
	launchID uint64,
) SourceOption {
	return func(c *Chain) {
		c.id = chainID
		c.url = sourceURL
		c.hash = sourceHash
		c.genesisURL = genesisURL
		c.genesisHash = genesisHash
		c.home = ChainHome(launchID)
	}
}
```